---
title: "Notebook with AMP-AD RNA-seq network analysis results"
output: html_notebook
---

Build aggregate modules from individual modules and differential expression meta-analysis, this will give you the same modules as are stored at syn11932957 - this code block builds the aggregate modules, and then checks if they are different from the paper aggregate modules.

```{r}
synapser::synLogin()
aggModsNew <- AMPAD::buildAggregateModules()
aggModsPaper <- synapser::synTableQuery('select * from syn11932957')$asDataFrame()[,-c(1,2)]

sum(aggModsNew[,1]!=aggModsPaper[,1])
sum(aggModsNew[,2]!=aggModsPaper[,2])
```


Make the pairwise enrichment plot (Figure 1B)

```{r}
synapser::synLogin()
AMPAD::pairwiseMatrixOfEnrichments('syn11932957')
AMPAD::pairwiseMatrixOfEnrichments('syn11932957',
                                   outputFile = T,
                                   fileName1 = '~/Desktop/updatedFiguresJan72020/figure1b.tiff')
```



Show that the aggregate modules are better enriched than other potential AD modules (Figure 1A)

```{r}
synapser::synLogin()
g <- AMPAD::improvedAdRelevancePlot()
g
```



Show summaries of overlaps with UpSet plots (Figure S1A-S1E)

```{r}
synapser::synLogin()
res<-AMPAD::aggregate_module_summary_plots()
res
```


Build aggregate modules without restricting the aggregation operation to the individual modules that are enriched for differentially expressed genes in AD.

```{r}
synapser::synLogin()
aggModsNonAdNew <- AMPAD::buildAggregateNonADModules()

```

Show overlap between AD and non AD (Figure S2A)
```{r}
AMPAD::compareADtoNonADmods(aggModsNew,
                            aggModsNonAdNew)

```




test building of gene sets

```{r}
targetedEnrichment <- AMPAD::run_full_enrichment_suite()
```

Run MAGMA analysis (must have magma tool installed in current working directory that this notebook is running from), producing the plot from Figure S1F
```{r}
synapser::synLogin()
g <- AMPAD::run_magma_analysis()
g
```


Run expression weighted cell enrichment method (must have downloaded and unzipped the flat file from GSE97930 in your current working directory: https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE97930&format=file&file=GSE97930%5FFrontalCortex%5FsnDrop%2Dseq%5FUMI%5FCount%5FMatrix%5F08%2D01%2D2017%2Etxt%2Egz )

This step also may require a large compute instance (with > 64 Gb of memory) to successfully run.

```{r}
synapser::synLogin()
g <- AMPAD::run_ewce()
g
```


Plot from Figure 1c
```{r}
synapser::synLogin()
modMeta <- AMPAD::getModuleMetainfo('syn11932957')
dummyDf <- targetedEnrichment$cell
dummyDf$fisherOR[dummyDf$adj.pval>=0.05] <- NA
#dummyDf <- dplyr::filter(targetedEnrichment$cell,adj.pval<=0.05)
dummyDf <- dplyr::left_join(dummyDf,modMeta)
dummyDf$category <- gsub('Zhang\\.','',dummyDf$category)
dummyDf$category <- factor(dummyDf$category,levels = rev(c('Astrocyte',
                                                       'Endothelial',
                                                       'Microglia',
                                                       'Neuron',
                                                       'MyelinOligos',
                                                       'NewOligos',
                                                       'OPC')))
g<-AMPAD::make_dot_plot(dummyDf,
                 xlab='AD Coexpression Module',
                 ylab='Cell Type Signature')
g
g<-AMPAD::make_dot_plot(dummyDf,
                 xlab='AD Coexpression Module',
                 ylab='Cell Type Signature',
                 outputFile = TRUE,
                 fileName = '~/Desktop/updatedFiguresJan72020/figure1c.tiff')
```

Plot from Figure S2B

```{r}
dummyDf <- targetedEnrichment$cell2
dummyDf$fisherOR[dummyDf$adj.pval>=0.05] <- NA
#dummyDf <- dplyr::filter(targetedEnrichment$cell,adj.pval<=0.05)
dummyDf <- dplyr::left_join(dummyDf,modMeta)

g<-AMPAD::make_dot_plot(dummyDf,
                 xlab='AD Coexpression Module',
                 ylab='Lake et al. Cell Type Signature')
g
g<-AMPAD::make_dot_plot(dummyDf,
                 xlab='AD Coexpression Module',
                 ylab='Lake et al. Cell Type Signature',
                 outputFile = TRUE,
                 fileName = '~/Desktop/updatedFiguresJan72020/figureS2b.tiff')
```

Plot for figure S6B

```{r}
dummyDf <- targetedEnrichment$scz
dummyDf <- dplyr::filter(dummyDf,category == 'yellow' | category == 'red' | category == 'blue' | category == 'lightyellow' | category == 'greenyellow' | category == 'cyan' | category == 'grey60')
dummyDf$fisherOR[dummyDf$adj.pval>=0.05] <- NA
#dummyDf <- dplyr::filter(targetedEnrichment$cell,adj.pval<=0.05)
dummyDf <- dplyr::left_join(dummyDf,modMeta)

g<-AMPAD::make_dot_plot(dummyDf,
                 xlab='AD Coexpression Module',
                 ylab='CommonMind Differentially Expressed Module')
g
g<-AMPAD::make_dot_plot(dummyDf,
                 xlab='AD Coexpression Module',
                 ylab='CommonMind Differentially Expressed Module',
                 outputFile = TRUE,
                 fileName = '~/Desktop/updatedFiguresJan72020/figures6b.tiff')


```

plot from figure S6C


```{r}
dummyDf <- targetedEnrichment$mssm2
dummyDf <- dplyr::filter(dummyDf,category == 'Yellow' | category == 'Pink' | category == 'Gray.1' | category == 'Seashell' | category == 'Red.3' | category == 'Green.yellow' | category == 'Red' | category == 'Gold.2' | category == 'Tan' | category == 'Gold.3' | category == 'Light.yellow' | category == 'Brown.2' | category == 'Dark.cyan' | category == 'Khaki' | category == 'Grey.60' | category == 'Purple' | category == 'Green.4' | category == 'Honey.dew' | category == 'Red.2' | category == 'Beige')
dummyDf$fisherOR[dummyDf$adj.pval>=0.05] <- NA
#dummyDf <- dplyr::filter(targetedEnrichment$cell,adj.pval<=0.05)
dummyDf <- dplyr::left_join(dummyDf,modMeta)
g<-AMPAD::make_dot_plot(dummyDf,
                 xlab='AD Coexpression Module',
                 ylab='Zhang et al. Module')
g

g<-AMPAD::make_dot_plot(dummyDf,
                 xlab='AD Coexpression Module',
                 ylab='Zhang et al. Module',
                 outputFile = TRUE,
                 fileName = '~/Desktop/updatedFiguresJan72020/figures6c.tiff')
```

Plot from figure 1D

```{r}
dummyDf <- targetedEnrichment$ad
dummyDf$fisherOR[dummyDf$adj.pval>=0.05] <- NA
dummyDf <- dplyr::left_join(dummyDf,modMeta)
#dummyDf$ModuleBrainRegion <- factor(dummyDf$ModuleBrainRegion,levels = unique(dummyDf$ModuleBrainRegion))
dummyDf <- dplyr::filter(dummyDf,category!='Nominated_targets' & category!='Mayo_simple' & category != 'omim' & category != 'jensenDisease' & category != 'wikipathwaysMouse' & category != 'pantherAmyloid')


dummyDf$category <- gsub('Mayo_comprehensive',
                         'Mayo Oligo Modules',
                         dummyDf$category)


dummyDf$category <- gsub('Columbia_Broad_Rush_m109',
                         'ROSMAP m109',
                         dummyDf$category)

dummyDf$category <- gsub('Emory',
                         'Emory AD Modules',
                         dummyDf$category)

dummyDf$category <- gsub('MSSM',
                         'MSSM Oligo Modules',
                         dummyDf$category)


# dummyDf$category <- factor(dummyDf$category,
#                            levels = rev(c('genecards',
#                                        'pantherPresenilin',
#                                        'dbgap',
#                                        'igap',
#                                        'jensenDisease',
#                                        'omimExpanded',
#                                        'biocarta',
#                                        'wikipathwaysMouse',
#                                        'wikipathwaysHuman',
#                                        'pantherAmyloid',
#                                        'kegg',                                       
#                                        'omim',
#                                        'Nominated_targets',
#                                        'MSSM',
#                                        'Mayo_simple',
#                                        'Mayo_comprehensive',
#                                        'Emory',
#                                        'Columbia_Broad_Rush_m109')))

dummyDf$category <- factor(dummyDf$category,
                           levels = rev(c('genecards',
                                       'pantherPresenilin',
                                       'dbgap',
                                       'igap',
                                       'omimExpanded',
                                       'biocarta',
                                       'wikipathwaysHuman',
                                       'kegg',                                       
                                       'MSSM',
                                       'Mayo_RNAseq',
                                       'Emory',
                                       'Columbia_Broad_Rush_m109')))


g<-AMPAD::make_dot_plot(dummyDf,
                 xlab='AD Coexpression Module',
                 ylab='AD Signature')
g

g<-AMPAD::make_dot_plot(dummyDf,
                 xlab='AD Coexpression Module',
                 ylab='AD Signature',
                 outputFile = TRUE,
                 fileName = '~/Desktop/updatedFiguresFeb52020/figure1d.tiff',
                 scaleValue = 2)
```

plot from figure 4D

```{r}
dummyDf <- dplyr::filter(targetedEnrichment$degMeta,adj.pval<=0.05)
dummyDf <- dplyr::filter(dummyDf,category == 'ad.control.FEMALE.random.DOWN' |
                           category == 'ad.control.FEMALE.random.UP' | 
                           category == 'ad.control.MALE.random.DOWN' | 
                           category == 'ad.control.MALE.random.UP')
dummyDf$adj.pval[dummyDf$adj.pval==0] = 10^-300
dummyDf$category <- gsub('ad\\.control\\.FEMALE\\.random\\.DOWN','Female, Down',dummyDf$category)
dummyDf$category <- gsub('ad\\.control\\.MALE\\.random\\.DOWN','Male, Down',dummyDf$category)
dummyDf$category <- gsub('ad\\.control\\.FEMALE\\.random\\.UP','Female, Up',dummyDf$category)
dummyDf$category <- gsub('ad\\.control\\.MALE\\.random\\.UP','Male, Up',dummyDf$category)
dummyDf <- dplyr::left_join(dummyDf,modMeta)

g<-AMPAD::make_dot_plot(dummyDf,
                 xlab='AD Coexpression Module',
                 ylab='DEG Sex Specific Meta Analysis')
g

g<-AMPAD::make_dot_plot(dummyDf,
                 xlab='AD Coexpression Module',
                 ylab='DEG Sex Specific Meta Analysis',
                 outputFile = TRUE,
                 fileName = '~/Desktop/updatedFiguresJan72020/figure4d.tiff')
```


